### DISCUSSION TITLE
python reasoner?

### DISCUSSION THREAD

**Discussion by doriantaylor** [Bioprotocols/labware-databank#4] [Q&A] [answered]
Just dropped by to ask ([via Tim Falllon](https://mastodon.social/@photocyte@genomic.social/110186143591148529)) what you folks are using as a reasoner in python?

**Comment by rpgoldman** [comment]
My understanding is that @markdoerr is using [Owlready](https://bitbucket.org/jibalamy/owlready2/src/master/).

**Comment by markdoerr** [comment, marked as answer]
Sorry Dorian (@doriantaylor), I was on travel ...

I am currently using Fact++, which is the default in EMMOntopy (https://github.com/emmo-repo/EMMOntoPy/tree/master/ontopy).
EMMOntopy is an addon package to owlready Robert (@rpgoldman ) mentioned with some nice extra features - and a very nice top-level ontology, coming for free.
I hope that answered your question.

**Reply by doriantaylor** [reply]
Great, thanks. Looking for a reasoner in Python that extends `rdflib`. CPython preferred. How do you square the Java stuff in OwlReady?

(Note: I'm aware there's something called FuXi but it's been abandonware for a while.)
